tools/src/main/scala/scala/scalanative/build/NativeLib.scala

```diff
@@ -166,33 +174,55 @@ private[scalanative] object NativeLib {
   /** Is this NativeLib in a jar file */
   private def isJar(nativelib: NativeLib): Boolean = isJar(nativelib.src)
 
+  private def srcExtensions(plugins: Seq[NativeSourcePlugin]): Seq[String] =
+    plugins.flatMap(_.extensions)
+
   /** Used to find native source files in jar files */
-  private val jarSrcRegex: String = {
-    val regexExtensions = srcExtensions.mkString("""(\""", """|\""", ")")
+  private def jarSrcRegex(plugins: Seq[NativeSourcePlugin]): String = {
+    val regexExtensions =
+      srcExtensions(plugins).mkString("""(\""", """|\""", ")")
```

Comment:
There are a lot of plugins passing here, maybe we move all of this method to a single `Impl` taking and plugins list. This might allow for some optimizations also. In most cases, we would probably create at most 2 instances per run, (1 instance per findNativeLibs and findNativePaths call)

Reply:
Oh yeah, that will be much better 😆